### PR TITLE
Added changelog fetching hook

### PR DESCRIPTION
**Pull request by rob-ghost**
ref https://linear.app/ghost/issue/BER-2610

## Why are we making this change?

The What's New feature needs to fetch and display Ghost changelog entries from `https://ghost.org/changelog.json`. This PR implements the changelog fetching logic as part of porting the What's New feature from Ember to React.

This PR builds on the test infrastructure established in #25356.

### Changelog Hook
1. **`useChangelog` hook**:
   - Fetches changelog data from `https://ghost.org/changelog.json`
   - Returns typed `ChangelogEntry[]` and `changelogUrl` string
   - Uses Zod for runtime schema validation
   - Gracefully handles missing/empty data with sensible defaults
   
2. **Type-safe changelog schema**:
   ```typescript
   type ChangelogEntry = {
     id: string;
     title: string;
     excerpt: string;
     url: string;
     publishedAt: Date;
     featured: boolean;
   };
   
   type ChangelogResponse = {
     entries: ChangelogEntry[];
     changelogUrl: string;
   };
   ```

3. **Robust error handling**:
   - HTTP errors (404, 500, etc.)
   - Network failures
   - Invalid JSON responses
   - Schema validation errors
   - All errors include helpful messages

### Test Infrastructure Addition
- **Changelog factory** (`test-utils/factories/changelog.ts`):
  - `createMockChangelogEntry()`: Creates typed changelog entries
  - `createRawChangelogEntry()`: Creates raw API response format
  - `createChangelogResponse()`: Creates full API response
  - `changelogFixtures`: Pre-configured test data for common scenarios

### Comprehensive Test Coverage
Tests cover:
- ✅ Successful data fetching and deserialization
- ✅ Missing/empty fields (posts, changelogUrl)
- ✅ HTTP errors (404, 500)
- ✅ Network failures
- ✅ Invalid JSON responses
- ✅ Schema validation errors (invalid URLs, dates, etc.)

## Why is this needed?

- **Decoupled from Ember**: React hook can be used independently
- **Type safety**: Zod validation prevents runtime errors from malformed API responses
- **Reusable**: Other features can use this hook to display changelog data
- **Foundation for What's New**: The What's New hook will combine this with user preferences

## Testing

```bash
yarn workspace @tryghost/admin test:unit src/whats-new/hooks/use-changelog.test.tsx
```

The tests use MSW to mock the Ghost changelog API and validate all edge cases.

## Dependencies

Depends on:
- #25356 (test infrastructure + user preferences)

### CODE REVIEW

File: apps/admin/src/whats-new/hooks/use-changelog.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    useQuery({
+        queryKey: ["changelog"],
+        queryFn: async () => {
+            const response = await fetch("https://ghost.org/changelog.json");
+
+            if (!response.ok) {
+                throw new Error(`Failed to fetch changelog: ${response.status}`);
+            }
+
+            const data = (await response.json()) as unknown;
```
Comment (jonatansberg) [inline apps/admin/src/whats-new/hooks/use-changelog.ts:52]: Nit: I think you can pass that type as a generic.

File: apps/admin/test-utils/factories/changelog.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export const createMockChangelogEntry = (overrides: Partial<ChangelogEntry> = {}): ChangelogEntry => ({
+    id: "1",
+    title: "Test Entry",
+    excerpt: "Test excerpt",
+    url: "https://ghost.org/changelog/1",
+    publishedAt: new Date("2025-01-15T10:00:00Z"),
+    featured: false,
+    ...overrides,
+});
```
Comment (jonatansberg) [inline apps/admin/test-utils/factories/changelog.ts]: This seems to be unused?

**Review by jonatansberg — COMMENTED** [review]
Looks good. I added a small nitpick comment. Also with the createMockChangelogEntry function in the factory seems to be unused.

File: apps/admin/src/whats-new/hooks/use-changelog.ts (hunk 1 above)
Comment (rob-ghost) [reply apps/admin/src/whats-new/hooks/use-changelog.ts:52]: Sorry I don't follow, could you give an example?

Comment (jonatansberg) [reply apps/admin/src/whats-new/hooks/use-changelog.ts:52]: I must have hallucinated this. Looking at the types, you can't do this. It might have been a type override that I've had in the past 🤔

File: apps/admin/test-utils/test-helpers.ts [hunk 3]
```diff
@@ -3,7 +3,11 @@ import { expect } from "vitest";
...
-    await waitFor(() => {
-        expect(result.current.isLoading).toBe(false);
-    });
+    await waitFor(
+        () => {
+            // Query is settled when it has reached a terminal state (success or error)
+            const isSettled = (result.current.isSuccess || result.current.isError) && !result.current.isFetching;
+            expect(isSettled).toBe(true);
+        }
+    );
```
[the remark this answers is not in the document]
Comment (rob-ghost) [reply apps/admin/test-utils/test-helpers.ts:12]: We actually want the opposite behaviour for this helper, we want idle queries to timeout since our expectation is that the query will eventually settle. There are not any tests which currently rely on disabled queries.

**Review by jonatansberg — APPROVED** [review]
This is good enough to merge. Any additional renames and refactors and tweaks can be handled as follow-ups.
